File: server/nn/event.py
from collections import namedtuple
from enum import Enum

Event = namedtuple('Event', ['log', 'timestamp', 'host', 'level', 'fields'])

class Level(Enum):
    DEBUG    = 7
    INFO     = 6
    NOTICE   = 5
    WARNING  = 4
    ERROR    = 3
    CRITICAL = 2
    ALERT    = 1

File: server/src/nnd/event/event.go
package event

import "time"

type Event struct {
	Log       string
	Timestamp time.Time
	Host      string
	Level     Level
	Fields    map[string]string
}

type Level int

const (
	DEBUG    Level = 7
	INFO     Level = 6
	NOTICE   Level = 5
	WARNING  Level = 4
	ERROR    Level = 3
	CRITICAL Level = 2
	ALERT    Level = 1
)
